---
title: Tutorial - introduction to METABRIC 
output: html_document
---
# Instructions

You have been provided with an R script to do some analysis. Please run this script step by step. Below is an explanation of the data, and each part of the code.

# Background 

This tutorial considers some exploratory analysis of data from METABRIC (Molecular Taxonomy of Breast Cancer International Consortium).

The data were originally used in a study that looked at the patterns of molecules inside tumours. Their conclusion was that ‘breast cancer’ is in fact at least ten different diseases, each with its own molecular fingerprint, and each with different weak spots. You can read about it here:

https://pubmed.ncbi.nlm.nih.gov/27161491/

The data you will analyse were downloaded from [cbioportal.org](https://www.cbioportal.org/study/clinicalData?id=brca_metabric)

# Analysis

The METABRIC data consist of clincal characteristics coupled with gene expression data. In this tutorial we look at the clinical characteristics. This is similar to a first stage in prognostic modelling, to understand the existing factors, before proceeding to investigate the new ones and how they relate to the existing ones.

Only a few simple analyses are done. For additional learning of R, I suggest that you try to conduct your own analysis of other factors.

# Script explanation

## Load the data

Data are already downloaded, and saved in the project (`brca_metabric_clinical_data.tsv`). This is a tab-separated file, and so `read.delim()` is used in place of `read.csv()`. `read.delim()` requires the user to specifiy *what* the separator is, and `"\t"` tells it that it is a tab.

```{r, echo=TRUE}
## 1. Load data
mydta = read.delim("brca_metabric_clinical_data.tsv", sep="\t")

## 2. Show first few rows, list column names
head(mydta)
```

## Summary statistics

```{r, echo=TRUE}
summary(mydta)
```

## Age at diagnosis

The below plots a histogram and boxplot of age on the same plot, with the histogram above the boxplot. The boxplot is orientated horizontally, rather than the default vertical boxplot.
```{r, echo=TRUE}
par(mfrow=c(2,1))

hist(mydta$Age.at.Diagnosis, xlab="Age", main="Age Histogram")

boxplot(mydta$Age.at.Diagnosis, xlab="Age", main="Age Boxplot", horizontal=TRUE)
```

## Lymph Nodes

The same plot as for age is repeated for number of lymph nodes. 

A difference is use of the `breaks=` argument. This tells R that the bins in the histogram should be with 1 (`by=1`), running from 0 to the maximum in the data (`max(ydta$Lymph.nodes.examined.positive, na.rm=TRUE)`. It is necessary to use the `na.rm=TRUE` argument in `max()` because this variable is not know for all women. 
```{r, echo=TRUE} 
par(mfrow=c(2,1))

hist(mydta$Lymph.nodes.examined.positive, breaks=seq(0,max(mydta$Lymph.nodes.examined.positive, na.rm=TRUE), by=1), col=1, main="")

boxplot(mydta$Lymph.nodes.examined.positive, xlab="Nodes", main="Nodes Boxplot", horizontal=TRUE)
```

If we wish to tabulate the number per category, then `table()` can be used as follows.

```{r, echo=TRUE}
table(mydta$Lymph.nodes.examined.positive)
```

## Tumour size

Next consider the same plots but for tumour size (in mm).

```{r, echo=TRUE}
par(mfrow=c(2,1))

hist(mydta$Tumor.Size, xlab="Tumour size", main="Tumour size Histogram")

boxplot(mydta$Tumor.Size, xlab="Tumour size", main="Tumour size Boxplot", horizontal=TRUE)
```

## Tumour size vs Lymph nodes

Finally, we look at the association between tumour size and lymph nodes. It is expected that larger tumours are more likely to have spread to other parts of the body, and so the two should be positively correlated. We examine their association next.

First, we plot the data

```{r, echo=TRUE}
plot(mydta$Tumor.Size, mydta$Lymph.nodes.examined.positive, log="x")

plot(mydta$Lymph.nodes.examined.positive,mydta$Tumor.Size, log="y")
```

This plots size vs nodes, and nodes vs size. We use the `log="x"` argument to put the x-axis on a log scale. This is sometimes useful for insepction of the plots, particularly when some data is a long way from the bulk.

Next let us define node positive vs negative in the data frame, and use this to get summary statistics of tumour size by nodal statue (neg/pos).
```{r, echo=TRUE} 
mydta$LNpos =  mydta$Lymph.nodes.examined.positive >0

tapply(mydta$Tumor.Size, mydta$LNpos, summary)
```

We can also plot the distribution of tumour size by nodal status as follows.

```{r, echo=TRUE}
sizeneg = subset(mydta$Tumor.Size, mydta$LNpos)

sizepos = subset(mydta$Tumor.Size, !mydta$LNpos)

par(mfrow=c(2,1))

hist(sizeneg, xlim=c(0,80), breaks=seq(0,200,by=5), col=1)

hist(sizepos, xlim=c(0,80), breaks=seq(0,200,by=5), col=1)
```

To test the hypothesis that mean tumour size is the same in both (independent) groups one might use a t-test, or a non-parametric Wilcoxon test.
```{r,echo=TRUE}
t.test(sizeneg, sizepos)

wilcox.test(sizeneg, sizepos)
```

One might sometimes be interested in particular cutpoints. For tumour size, 20mm is a cutpoint used in tumour staging, so we look at that next.

First we calculate the number sized more than 20mm positive by nodal status 

```{r, echo=TRUE}
npos = c(sum(sizeneg>20, na.rm=TRUE), sum(sizepos>20, na.rm=TRUE))

ntot = c(sum(!is.na(sizeneg)), sum(!is.na(sizepos)))

## number size >20mm
npos

## number with size available
ntot

## proportion
npos / ntot
```

Then we can carry a test of the difference in two proportions (based on a z-test) as follows

```{r, echo=TRUE}
prop.test(npos, ntot)
```

# Now your turn..

Keep exploring!
